src/goto-programs/slice_global_inits.cpp

```diff
@@ -0,0 +1,143 @@
+/*******************************************************************\
+
+Module: Remove initializations of unused global variables
+
+Author: Daniel Poetzl
+
+Date:
+
+\*******************************************************************/
+
+#include <analyses/call_graph.h>
+
+#include <util/namespace.h>
+#include <util/std_expr.h>
+#include <util/cprover_prefix.h>
+#include <util/prefix.h>
+
+#include <goto-programs/goto_functions.h>
+
+#include "slice_global_inits.h"
+
+/*******************************************************************\
+
+Function: slice_global_inits
+
+  Inputs:
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
+
+void slice_global_inits(
+  const namespacet &ns,
+  goto_functionst &goto_functions)
+{
+  // gather all functions reachable from the entry point
+
+  call_grapht call_graph(goto_functions);
+  const call_grapht::grapht &graph=call_graph.graph;
+
+  std::list<irep_idt> worklist;
+  std::set<irep_idt> functions_reached;
```

Comment:
In several places in this file: use hash_set_cont<irep_idt, irep_id_hash> instead of std::set